### ISSUE TITLE
Add the ability to hide the fence badges

### ISSUE THREAD

**Issue by TheoChevalier** [mozilla/contain-facebook#395]
Hi there,

Is there a way to hide the fence badges in v1.9 without disabling the add-on entirely (even by setting a pref value)? I’m hitting a few cases where the badges are displayed and it’s a big no-no :/ 

1. Some French catch-up TV websites, like https://www.6play.fr

Badges stay over the content while streaming with no way to remove them
![Screenshot_2019-05-23 19 45 du mercredi 22 mai - 6play](https://user-images.githubusercontent.com/1294206/58247445-fd7e8780-7d59-11e9-90a8-939b8fb9e1aa.png)

2. ProtonMail. When you open an email containing a Facebook tracker, the badge gets displayed, and even if you open other emails, it stays in the way.

![Screenshot_2019-05-23 (312) Boîte de réception ProtonMail](https://user-images.githubusercontent.com/1294206/58247889-fdcb5280-7d5a-11e9-80c5-9e35eb4dd94d.png)

I’m assuming I’ll hit other cases like this, so I think allowing people to hide them would be great.

Thanks!

**Comment by maxxcrawford** [comment]
Yes! We’re working internally on implementing this in one of the next immediate releases. We should land something as early as the next two weeks. 😸

**Comment by AeonEternum** [comment]
Gets more interesting on imgur.com: https://i.imgur.com/UWKfhyc.jpg
Thank you!

**Comment by TheUnixDude** [comment]
Any update on this?

**Comment by maxxcrawford** [comment]
@TheUnixDude Our 2.0 release included only showing one single badge per page and it only displays on visible items. This isn't a direct solution to this issue, but it does solve some use cases. Do you have a specific site with this issue?

**Comment by TheUnixDude** [comment]
Thanks for the reply.

(Firefox 68.0.1, Debian Stable, plugin version 2.0.1) 

On pretty much any page I visit which both triggers the plugin and has content requiring lots of vertical scrolling (think: reddit, various new sites, imgur, or similar), the scrolling causes the icon to hover over various parts of the page; it's a good 1-2 seconds of activity behind whatever I'm doing, as if the icon refresh rate is 1s or so. Eventually it will settle on the offending item in the page if I stop scrolling.

I'd prefer to simply not have any visible indication in the content at all - if the plugin works, I don't need a red icon bouncing over my content reminding me. A simple color change of the plugin icon in the Firefox bar (ala UBlock Origin, Privacy Badger, etc) would be more than sufficient.

**Comment by kparal** [comment]
I also find the badge unnecessarily distracting, especially during scrolling. I don't see the point of the badge, except for self-promoting (it's the same reason why software (think antivirus, etc) often keep an animated icon in Windows systray during almost any activity - it has almost zero informational value for users and just distracts them, but makes sure they are aware of the product, therefore strengthening the brand). I installed the extension and expect it to work. I want to know when it doesn't work for some reason, and not every time it does. I would very much prefer to disable the badge completely. Thank you for consideration. (Please note that I very much appreciate you working on this. It's just the current presentation that I don't enjoy).

**Comment by AeonEternum** [comment]
I swear I read somewhere that a toggle is planned for the icon.

> Gets more interesting on imgur.com: https://i.imgur.com/UWKfhyc.jpg
> Thank you!

Anyhow, I left this comment on May 28th and it has since gotten better. In most cases I see either none or only 1 icon when browsing imgur.

**Comment by theprojectsomething** [comment]
Just to add to this: IMO leaving the current implementation as the default setting is fine; it's important new installs visually showcase how the add-on works (in-situ reminders of broken features are important, and especially for new Firefox users who may have come across on the back of the privacy campaign). However, allowing for configuration

**Comment by Pfaufisch** [comment]
> @TheUnixDude Our [latest release](https://github.com/mozilla/contain-facebook/releases/tag/2.0.1) included only showing one single badge per page and it only displays on visible items. This isn't a direct solution to this issue, but it does solve some use cases. Do you have a specific site with this issue?

Video player of the New York Times is broken. Any interaction with the player (turning on HD, pressing play/pause) has the icon appearing and sticking on top of the player, even when the UI fades out.

![Bildschirmfoto 2019-12-11 um 22 42 18](https://user-images.githubusercontent.com/10010637/70663083-e2c51f00-1c67-11ea-9834-5a9f9a6532ee.png)

**Comment by maxxcrawford** [comment]
Feature for #550
